navigation.py

```diff
@@ -42,12 +42,40 @@ def navigate_to_link(link, driver):
     # wait for page to load, up to ten seconds
     WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, "//html")))
 
-def assert_current_page_is(page_name, driver):
-    if page_name in pages:
-        assert driver.current_url == pages[page_name]
-    else:
+def assert_current_page_is(page_name, driver, retry=0):
+    """ Assert that current url is as given.
+
+    Option arguments:
+        retry -- specifies how often to retry, with one second of delay between retries
+    """
+    if page_name not in pages:
         raise PageNotImplementedError(page_name, pages)
 
+    target_url = pages[page_name]
+    assert 0 <= retry
```

Comment:
Is this necessary? If the assertion fails you'll just drop through the for loop and end up at the URL check, which will probably fail and give a more descriptive error message

Reply:
Dropped this, I wasn't aware that `range(-n)` is actually empty and not `[0, -1, -2, ...]`